lib/widgets/backup_in_progress_dialog.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:async';
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:c_breez/bloc/backup/backup_state.dart';
+import 'package:c_breez/routes/home/widgets/annimated_loader_dialog.dart';
+import 'package:flutter/material.dart';
+
+class BackupInProgressDialog extends StatefulWidget {
+  final Stream<BackupState?> backupBlocStream;
+
+  const BackupInProgressDialog(this.backupBlocStream);
+
+  @override
+  createState() => BackupInProgressDialogState();
+}
+
+class BackupInProgressDialogState extends State<BackupInProgressDialog> {
+  late StreamSubscription<BackupState?> _stateSubscription;
+  @override
+  void didChangeDependencies() {
```

Comment:
I think this dialog can be simplified by only focusing on the build function and not needing to implement the didChangeDependencies at all.
Can use BlocListener (https://pub.dev/documentation/flutter_bloc/latest/flutter_bloc/BlocListener-class.html)
To determine the right backup status and pop the widget off the stack when needed.